src/mumble/MainWindow.cpp

```diff
@@ -200,6 +202,27 @@ MainWindow::MainWindow(QWidget *p)
 	QAccessible::installFactory(AccessibleSlider::semanticSliderFactory);
 }
 
+// Loading a state that was stored by a different version of Qt can lead to a crash.
+// This function calculates the state version based on Qt's version and MainWindow.ui's hash (provided through CMake).
+// That way we also avoid potentially causing bugs/glitches when there are changes to MainWindow's widgets.
+int MainWindow::stateVersion() {
+	constexpr size_t hashLength = sizeof(MAINWINDOW_UI_HASH);
+	int version = 0;
+
+	for (size_t i = 0; i < hashLength / 2 * 2; i += 2) {
```

Comment:
What are you doing here? A comment would likely be helpful 👀

Reply:
It's simply to ensure that we don't read out of bounds, in case the hash is invalid/corrupt.

Reply:
It does look funny though so a comment explaining that dividing by two and then multiplying by two does not (necessarily) yield back the original number seems appropriate